control-plane/pkg/prober/cache.go

```diff
@@ -52,43 +52,45 @@ func (s Status) String() string {
 }
 
 // Cache is a key-status store.
-type Cache interface {
+type Cache[K comparable, V, A interface{}] interface {
 	// GetStatus retries the status associated with the given key.
-	GetStatus(key string) Status
+	Get(key K) (V, bool)
 	// UpsertStatus add or updates the status associated with the given key.
 	// Once the given key expires the onExpired callback will be called passing the arg parameter.
-	UpsertStatus(key string, status Status, arg interface{}, onExpired ExpirationFunc)
+	UpsertStatus(key K, value V, arg A, onExpired ExpirationFunc[K, V, A])
 	// Expire will expire the given key.
-	Expire(key string)
+	Expire(key K)
+	// Keys returns all of the keys currently in the cache that have not expired
+	Keys() []K
```

Comment:
This is unused, why do we need this method?

Reply:
Good catch, this was from an earlier version of the PR. Let me remove it